src/main/java/Duke.java

```diff
@@ -1,10 +1,138 @@
+        String description = eventTask.substring(0, slot - 1);
+        Event newItem = new Event(description, eventTask.substring(slot + 4));
+        tsk[Task.getTaskCount() - 1] = newItem;
+
+        System.out.println("\t" + "Got it. I've added this to your custom-list: ");
+        System.out.println("\t\t" + tsk[Task.getTaskCount() - 1].toString());
+        System.out.println("\t" + "Now you have " + Task.getTaskCount() + " tasks in your list :)");
+    }
+
+    private static void addDeadline(Task[] tsk, String deadlineTask) {
+        int deadline = deadlineTask.indexOf("/");
+        String description = deadlineTask.substring(0, deadline - 1);
+        Deadline newItem = new Deadline(description, deadlineTask.substring(deadline + 4));
+        tsk[Task.getTaskCount() - 1] = newItem;
+
+        System.out.println("\t" + "Got it. I've added this to your custom-list: ");
+        System.out.println("\t\t" + tsk[Task.getTaskCount() - 1].toString());
+        System.out.println("\t" + "Now you have " + Task.getTaskCount() + " tasks in your list :)");
+    }
+
+    private static void addToDo(Task[] tsk, String input) {
+        ToDo newItem = new ToDo(input);
+        tsk[Task.getTaskCount() - 1] = newItem;
+        System.out.println("\t Got it. I've added this to your custom-list: ");
+        System.out.println("\t\t" + tsk[Task.getTaskCount() - 1].toString());
+        System.out.println("\t Now you have " + Task.getTaskCount() + " tasks in your list :)");
+    }
+
+    private static void taskDone(Task[] tsk, String eventDone) {
+        int index = extractTaskNumber(eventDone);
+        if (index == 0 | index > Task.getTaskCount()) {
+            System.out.println("\t" + "Invalid task number");
+        } else {
+            tsk[index - 1].markAsDone();
+            Task.markTaskCompleted();
+            System.out.println("\t" + "Nice! I've marked this task as done:");
+            System.out.println("\t\t" + tsk[index - 1].toString());
+            System.out.println("\t" + "Type \"list\" to see a list of pending tasks");
+        }
+    }
+
+    private static void addTask(Task[] tsk, String taskDescription) {
+        Task newTask = new Task(taskDescription);
+        tsk[Task.getTaskCount() - 1] = newTask;
+    }
+
+    private static void listTasks(Task[] tsk) {
+        int j;
+        if (Task.getTaskCount() == 0) {
+            System.out.println("\t" + "You currently have no tasks");
+            System.out.println("\t" + "To update your to-do list, just type the task");
+        } else {
+            System.out.println("\t" + "Here are the tasks in your list:");
+            for (j = 0; j < Task.getTaskCount(); j++) {
+                System.out.println("\t" + (j + 1) + ". " + tsk[j].toString());
+            }
+        }
+    }
+
+    private static int extractTaskNumber(String command) {
+        int itemNo;
+        try {
+            itemNo = Integer.parseInt(command);
+        } catch (NumberFormatException e) {
```

Comment:
Coding standard followed well in this error handling portion of the function! Nice.